forge-gui/res/cardsfolder/r/repopulate.txt

```diff
@@ -1,7 +1,7 @@
 Name:Repopulate
 ManaCost:1 G
 Types:Instant
-A:SP$ ChangeZoneAll | Origin$ Graveyard | Destination$ Library | ValidTgts$ Player | TgtPrompt$ Select target player | ChangeType$ Creature | Shuffle$ True | SpellDescription$ Shuffle all creature cards from target player's graveyard into that player's library.
+A:SP$ ChangeZoneAll | Origin$ Graveyard | Destination$ Library | ValidTgts$ Player | TgtPrompt$ Select target player | ChangeType$ Creature | Shuffle$ True | AILogic$ ManifestCreatsFromGraveyard | SpellDescription$ Shuffle all creature cards from target player's graveyard into that player's library.
```

Comment:
thanks, I suppose this does pass the test but just applying the logic from _Ghastly Conscription_ doesn't really make sense since the cards are pretty different in their usage (e.g. the AI might now use this to save their opponent who's getting milled out :P)

you'd have to debug `ChangeZoneAllAi` to find out where the wrong play decision comes from
since there already appears to be some general logic for Graveyard as origin in place ideally that can be extended

as coding custom AILogic$ is usually reserved for more exotic cases (also depends a bit how many closely-related cards exist)

Reply:
right, it already uses the "ChangeType" param to find the best target before
but that check doesn't really take that into account

this leads to the question if there's a case where that behavior could be desirable 🤔
if there's no good reason for it often the answer is just that this API just wasn't fleshed out enough yet
(like how it currently ignores targeting yourself anyway)